src/pageServices/contentScripts/generationData.js

```diff
@@ -66,18 +66,32 @@ export const getGenerationAttributes = () => {
             */
   };
 
-  const generateSelectorByElement = (element) => {
+  const generateSelectorByElement = async (element) => {
     const generatorOptions = {
       blacklist: [/jdn-hash/, /href/],
     };
     const selectorByGenerator = getCssSelector(element, generatorOptions);
 
-    const finderForbiddenAttributes = ["jdn-hash", "href", "class", "xmlns", "xmlns:xlink", "xlink:href"];
-    const selectorByFinder = finder(element, {
-      attr: (name) => !finderForbiddenAttributes.includes(name),
-    });
-
-    return selectorByGenerator.length < selectorByFinder.length ? selectorByGenerator : selectorByFinder;
+    // Due to finder lib issue, we have to catch errors.
+    // If css selector generated by finder contains ".", for example #Country_16856319000360.6264611129794591
+    // We get "Uncaught DOMException: Failed to execute 'querySelectorAll' on 'Document':
+    // '#Country_16856319000360.6264611129794591' is not a valid selector."
+    // And because of lib logic we can't prevent generation these selectors.
+    // Reproduced on https://www.docker.com
+    try {
+      const finderForbiddenAttributes = ["jdn-hash", "href", "class", "xmlns", "xmlns:xlink", "xlink:href"];
+      const selectorByFinder = await finder(element, {
+        attr: (name, value) => {
+          console.log(name);
+          console.log(value);
+          return value && !value.includes(".") && !finderForbiddenAttributes.includes(name);
```

Comment:
if `!value.includes(".")` doesn't work, no need to keep this code

Reply:
removed